glommio/src/executor/mod.rs

```diff
@@ -29,6 +29,12 @@
 
 #![warn(missing_docs, missing_debug_implementations)]
 
+mod latch;
+mod placement;
```

Comment:
I don't mind moving executor.rs to a directory, but two things:

1. it's a conflict magnet, so I'd recommend doing that separately in a PR that can go in quickly. There are already conflicts due to recent changes in executor.rs, so be careful to carry them over.
2. please move multitask.rs as well, since it's only purpose is to isolate some of the functions of the executor. There may even be a case for moving parking.rs as well, although I am a bit more on the fence with that one, as there are functions from parking.rs used all over the place.